cdap-common/bin/functions.sh

```diff
@@ -440,11 +440,11 @@ cdap_set_hive_classpath() {
     # If Hive classpath is successfully determined, derive explore
     # classpath from it and export it to use it in the launch command
     if [[ -n ${HIVE_HOME} ]] && [[ -n ${HIVE_CONF_DIR} ]] && [[ -n ${HADOOP_CONF_DIR} ]]; then
-      EXPLORE_CONF_FILES=$(ls -1dF ${HIVE_CONF_DIR}/* ${HADOOP_CONF_DIR}/* | sed -e '/\/$/d' | tr '\n' ':')
-      EXPLORE_CLASSPATH=$(ls -1 ${HIVE_HOME}/lib/hive-exec-* ${HIVE_HOME}/lib/*.jar | tr '\n' ':')
+      EXPLORE_CONF_FILES="${HIVE_CONF_DIR}:${HADOOP_CONF_DIR}"
+      EXPLORE_CLASSPATH=${HIVE_CLASSPATH:-$(echo -e "${HIVE_VARS}" | grep '^env:CLASSPATH=' | cut -d= -f2)}
```

Comment:
Also, I don't think we can skip running `hive set` since we use it for extracting other information, like `hive.execution.engine`, `HIVE_CONF_DIR`, etc.